pgosupport.cmake

```diff
@@ -0,0 +1,68 @@
+function(clr_pgo_unknown_arch)
+    if (WIN32)
+        message(FATAL_ERROR "Only AMD64, ARM and I386 are supported for PGO")
+    else()
+        message(FATAL_ERROR "PGO not currently supported on the current platform")
+    endif()
+endfunction(clr_pgo_unknown_arch)
+
+# Adds Profile Guided Optimization (PGO) flags to the current target
+function(add_pgo TargetName)
+    if(WIN32)
+        set(ProfileFileName "${TargetName}.pgd")
+    endif(WIN32)
+
+    file(TO_NATIVE_PATH
+        "$ENV{__PackagesDir}/Microsoft.DotNet.OptimizationData.Coreclr/$ENV{__BuildOS}.$ENV{__BuildArch}/${ProfileFileName}"
+        ProfilePath
+    )
+
+    # Enable PGO only for optimized configs
+    set(ConfigTypeList RELEASE RELWITHDEBINFO)
+
+    foreach(ConfigType IN LISTS ConfigTypeList)
```

Comment:
@janvorli I originally tried this approach, but apparently, cmake does not expand generator expressions for `LINK_FLAGS*`. The best workaround I could find suggested using `target_link_libraries` instead, which kind of works, but on Windows it dumps the switches into `<AdditionalDependencies>` (next to all the \*.lib files) instead of `<AdditionalOptions>`, and this to me seems wrong (although cmake "supports" that, we're relying on undocumented msbuild behavior there).

I could try using `target_link_libraries` anyway; or I could just target `LINK_FLAGS_RELEASE` (and simply ignore PGO for checked builds). I'm kind of leaning toward the latter here, but I'm open to it either way... what do you think? (@brianrob, @gkhanna79 - any additional thoughts?)